cardano-db-sync/src/Cardano/DbSync/Default.hs

```diff
@@ -146,6 +146,8 @@ insertBlock syncEnv cblk applyRes firstAfterRollback tookSnapshot = do
           Generic.fromBabbageBlock (ioPlutusExtra iopts) (getPrices applyResult) blk
   insertEpoch details
   lift $ commitOrIndexes withinTwoMin withinHalfHour
+  when (unBlockNo blkNo `mod` 21600 == 0) $ do
```

Comment:
think that's meant to be `2160` not `21600`.

I ended up using something like bellow rather than hard coding this value:

```haskell
getSecurityParam :: SyncEnv -> Word64
getSecurityParam syncEnv =
  case envLedgerEnv syncEnv of
    HasLedger hle -> getSecurityParameter $ leProtocolInfo hle
    NoLedger nle -> getSecurityParameter $ nleProtocolInfo nle          
```
not sure that would be of use 🤷

Reply:
21600, or 10 * securityParameter, is the expected number of blocks per epoch on mainnet, so by doing this periodicaly every 21600 it's done once per epoch.
Initially I wanted to do this pruning at the first block of the epoch, however the block itself doesn't carry this information, it requires the ledger state and we have options that disable it. So this was the easiest check I can make to have a realistic benchmark just by looking at the block itself (no additional state).